FreeRTOS/FreeRTOS-Kernel-Partner-Supported-Ports
# Update Xtensa FreeRTOS support to v11

Bring Xtensa port up to date with FreeRTOS version 11, moving to Partner-Supported-Ports submodule.

Description
-----------
Bring existing Cadence-maintained Xtensa v10.4.4-stable port up to date in the FreeRTOS v11 context:
- Incorporate and consolidate all existing Xtensa port bug-fixes from internal, external, and third-party repositories
- Additional improvements in tickless-idle support, newlib, overlays, coprocessor management, and NX MPU and syscall handling.
- Maintain consistent formatting, copyright, and whitespace
- Update Xtensa port sub-version to 3.0.0

Test Steps
-----------
- Tested via various demo and stress tests as provided through the Partner-Supported-Demos submodule.
- Regression testing on many different LX7, LX8, and NX configurations.

Related Issue
-----------
N/A


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
